.github/workflows/publish-master-merges.yaml

```diff
@@ -0,0 +1,54 @@
+name: Publish all merges to master
+
+on:
+  push:
+    branches: ["master"]
+    paths:
+      - helm-chart/**
+      - scripts/**
+      - acceptance-tests/**
+      - .github/**
+  workflow_dispatch:
+
+jobs:
+  publish-chart:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions-ecosystem/action-get-latest-tag@v1
+        id: get-latest-tag
+      - uses: actions-ecosystem/action-bump-semver@v1
+        id: bump-semver
+        with:
+          current_version: ${{ steps.get-latest-tag.outputs.tag }}
+          level: prepatch
+      - uses: SwissDataScienceCenter/renku-actions/publish-chart@v0.3.0
+        env:
+          CHART_DIR: helm-chart/
+          CHART_TAG: "--tag ${{ steps.bump-semver.outputs.new_version }}"
```

Comment:
I think it's better to just let chartpress set the default tag - that way it's always identifiable by the commit hash. Our charts used to have dev versions using the `-<commit-sha-7>` suffix - that can also work here I think and it would be consistent with the old charts (not that it's a big requirement in my mind)

Reply:
There are two reasons to not let chartpress do its things:

1. Chartpress does not understand Semver: it uses the **latest** stable tag as base for a new pre-release (e.g. in this case would call it `0.10.3-xxx`, this is wrong because it would tag a pre-release for something that has been already released. This is not only a problem of being pedantic in the use of Semver: since Flux implements correctly the versioning it will never deploy `0.10.3-xxx` because `0.10.3` **is more recent** and has [precedence](https://semver.org/#spec-item-11)
2. By using the commit hash as pre-release version we don't know anymore what is the correct ordering of pre-releases and thus which one to upgrade. It is possible to maintain the hash in the pre-release version by [appending](https://semver.org/#spec-item-10) it with a `+`, e.g. `0.10.4-2+a6a1f62`.

Because of point 2 we also have to push the tag back in this repository so that the workflow knows which one was the latest pre-release published and can bump the new one correctly.

Reply:
yeah I see what you mean - the behavior of chartpress is incorrect in this regard, or rather, `git describe --tags` doesn't produce semver-compliant output. 

If I run `git describe --tags` on current master, I get 

```
0.10.3-3-gb696ff7d
```

which means I am three commits after `0.10.3` at commit `b696ff7d` - can't we do something like this, but use proper semver, making this simply `0.10.4-3+gb696ff7d`? That way automation which requires semver for ordering can still use it and we don't need to push this tag back to the repo since it's easy to recreate.

Reply:
yes, I've mentioned it in the reply above :)

Reply:
right, I was agreeing with you - but I don't think we need to tag every commit to master since it's easy to build that part of the version just from git history.

Reply:
FYI the PR is now ready to review, the final implementation turned out to be `x.y.z-n.<short_commit_hash>` to not have to fiddle too much with Semver manipulation.